shared/chat/conversation/messages/attachment.desktop.js

```diff
@@ -1,69 +1,195 @@
 }
 
-// TODO abstract this part so it is the same as message text
-class _AttachmentMessage extends PureComponent<void, Props & {onIconClick: (event: any) => void}, void> {
-  render () {
-    const {message, style, includeHeader, isFirstNewMessage, onLoadAttachment, onOpenInFileUI, onIconClick} = this.props
-    const {downloadedPath} = message
+function PreviewImage ({message: {previewPath, previewType, messageState}}: {message: Constants.AttachmentMessage}) {
+  if (!!previewPath && previewType === 'Image') {
+    let style = {
+      ...globalStyles.flexBoxRow,
+      marginTop: globalMargins.xtiny,
+      flex: 1,
+      position: 'relative',
+      alignItems: 'flex-end',
+    }
+    switch (messageState) {
+      case 'uploading':
+      case 'pending':
+      case 'failed':
+        style = {...style, opacity: 0.4}
+        break
+    }
+
     return (
-      <Box style={{...globalStyles.flexBoxColumn, flex: 1, ...(isFirstNewMessage ? stylesFirstNewMessage : null), ...style}} className='message'>
-        <Box style={{...globalStyles.flexBoxRow, flex: 1}}>
-          <Box style={{width: 2, marginRight: globalMargins.tiny, alignSelf: 'stretch', backgroundColor: _marginColor(message.followState)}} />
-          <Box style={{...globalStyles.flexBoxRow, flex: 1, paddingTop: (includeHeader ? globalMargins.tiny : 0)}}>
-            {includeHeader
-              ? <Avatar size={24} username={message.author} style={{marginRight: globalMargins.tiny}} />
-              : <Box style={{width: 32}} />}
-            <Box style={{...globalStyles.flexBoxColumn, flex: 1}}>
-              {includeHeader && <Text type='BodySmallSemibold' style={{color: colorForAuthor(message.followState), ...(message.followState === 'You' ? globalStyles.italic : null)}}>{message.author}</Text>}
-              <Box style={{...globalStyles.flexBoxRow, flex: 1}}>
-                {!message.downloadedPath &&
-                  <Text type='Body' style={{marginTop: globalMargins.xtiny, flex: 1}} onClick={() => onLoadAttachment(message.messageID, message.filename)}>
-                    Click to download: {message.title} - {message.filename}
-                  </Text>}
-                {!!downloadedPath &&
-                  <Text type='Body' style={{marginTop: globalMargins.xtiny, flex: 1}} onClick={() => onOpenInFileUI(downloadedPath)}>
-                    Show downloaded file.
-                  </Text>}
-                {!!message.previewPath && message.previewType === 'Image' && <Box style={{marginTop: globalMargins.xtiny, flex: 1}}><img src={message.previewPath} /></Box>}
-                <div className='action-button'>
-                  <Icon type='iconfont-ellipsis' style={{marginLeft: globalMargins.tiny, marginRight: globalMargins.tiny}} onClick={onIconClick} />
-                </div>
-              </Box>
-            </Box>
-          </Box>
-        </Box>
+      <Box style={style}>
+        <img style={{...globalStyles.rounded}} src={previewPath} />
```

Comment:
i believe you should have this info. the attachment metadata has the image sizes i think @patrickxb ?